src/main/java/com/faforever/client/chat/ChatUserService.java

```diff
@@ -174,30 +167,60 @@ public void associatePlayerToChatUser(ChatChannelUser chatChannelUser, Player pl
       populateCountry(chatChannelUser);
       populateAvatar(chatChannelUser);
       populateColor(chatChannelUser);
-      chatChannelUser.setAvatarInvalidationListener((observable) -> populateAvatar(chatChannelUser));
-      chatChannelUser.setClanTagInvalidationListener((observable) -> populateClan(chatChannelUser));
-      chatChannelUser.setCountryInvalidationListener((observable) -> populateCountry(chatChannelUser));
-      chatChannelUser.setSocialStatusInvalidationListener((observable) -> populateColor(chatChannelUser));
-      chatChannelUser.setGameStatusInvalidationListener((observable) -> populateGameImages(chatChannelUser));
-      chatChannelUser.setPopulatedInvalidationListener((observable -> {
-        populateGameImages(chatChannelUser);
-        populateClan(chatChannelUser);
-        populateCountry(chatChannelUser);
-        populateAvatar(chatChannelUser);
-        populateColor(chatChannelUser);
-      }));
+      chatChannelUser.setAvatarChangeListener((observable, oldValue, newValue) -> {
+        if (!oldValue.equals(newValue)) {
+          populateAvatar(chatChannelUser);
+        }
+      });
+      chatChannelUser.setClanTagChangeListener((observable, oldValue, newValue) -> {
+        if (!oldValue.equals(newValue)) {
+          populateClan(chatChannelUser);
+        }
+      });
+      chatChannelUser.setCountryChangeListener((observable, oldValue, newValue) -> {
+        if (!oldValue.equals(newValue)) {
+          populateCountry(chatChannelUser);
+        }
+      });
+      chatChannelUser.setSocialStatusChangeListener((observable, oldValue, newValue) -> {
+        if (!oldValue.equals(newValue)) {
+          populateColor(chatChannelUser);
+        }
+      });
+      chatChannelUser.setGameStatusChangeListener((observable, oldValue, newValue) -> {
+        if (!oldValue.equals(newValue)) {
+          populateGameImages(chatChannelUser);
+        }
+      });
```

Comment:
This throws NPE if `oldValue` is null. You can use `Objects.equals(oldValue, newValue)` to avoid this

Reply:
good catch